quicktree.php

```diff
@@ -4,226 +4,406 @@
+			array($user));
+
+		if (cacti_sizeof($graphs)) {
+			include_once($config['base_path'] . '/lib/api_tree.php');
+
+			if (empty($new_tree_id)) {
+				$seq = db_fetch_cell('SELECT MAX(sequence) FROM graph_tree');
+
+				if ($seq == NULL || $seq < 0) {
+					$seq = 1;
+				}
+
+				$save = array();
+				$save['id']            = '';
+				$save['name']          = $tree_name;
+				$save['sort_type']     = TREE_ORDERING_ALPHABETIC;
+				$save['sequence']      = $seq;
+				$save['last_modified'] = date('Y-m-d H:i:s', time());
+				$save['modified_by']   = $_SESSION['sess_user_id'];
+
+				if (empty($save['id'])) {
+					$save['user_id'] = $_SESSION['sess_user_id'];
+				}
+
+				$new_tree_id = sql_save($save, 'graph_tree');
+
+				if ($branch != '') {
+					$parent_id = api_tree_item_save(0, $new_tree_id, TREE_ITEM_TYPE_HEADER,
+						0, $branch, 0, 0, 0, 1, TREE_ORDERING_INHERIT, false);
+				} else {
+					$parent_id = 0;
+				}
+			} else {
+				// if an existing tree was picked, create a new heading to
+				// be the parent for all our graphs
+				$parent_id = db_fetch_cell_prepared('SELECT id FROM graph_tree_items
+					WHERE graph_tree_id = ?
+					AND title = ?
+					AND local_graph_id = 0
+					ORDER BY id
+					LIMIT 1',
+					array($new_tree_id, $branch));
+
+				if (empty($parent_id)) {
+					$parent_id = api_tree_item_save(0, $new_tree_id, TREE_ITEM_TYPE_HEADER,
+						0, $branch, 0, 0, 0, 1, TREE_ORDERING_INHERIT, false);
+				}
+			}
+
+			foreach ($graphs as $gr) {
+				$tree_item_id = api_tree_item_save(0, $new_tree_id, TREE_ITEM_TYPE_GRAPH,
+					$parent_id, '', $gr['local_graph_id'], $gr['rra_id'], 0, 1, TREE_ORDERING_INHERIT, false);
+			}
+
+			raise_message('tree_done', __('QuickTree has Created Tree and/or Branch and added Graphs', 'quicktree'), MESSAGE_LEVEL_INFO);
+
+			if ($parent_id > 0) {
+				$url = $config['url_path'] . 'graph_view.php?action=tree&node=tbranch-' . $parent_id . '&site_id=-1&host_id=-1&host_template_id=-1&hgd=&hyper=true';
+			} else {
+				$url = $config['url_path'] . 'graph_view.php?action=tree&node=tree_anchor-' . $new_tree_id . '&site_id=-1&host_id=-1&host_template_id=-1&hgd=&hyper=true';
+			}
+
+			?>
+			<script type="text/javascript"> $(function() { document.location = "<?php print $url;?>"; }); </script>
+			<?php
+
+			exit;
+		} else {
+			raise_message('nographs', __('QuickTree has no Graphs Queued', 'quicktree'), MESSAGE_LEVEL_ERROR);
+
+			header('Location: ' . $config['url_path'] . 'plugins/quicktree/quicktree.php');
+			exit;
+		}
+
+		break;
+	case 'remove':
+		$graph = 0;
+
+		if (isset_request_var('id')) {
+			$graph = get_filter_request_var('id');
+
+			$result = db_execute_prepared('DELETE FROM quicktree_graphs
+				WHERE userid = ?
+				AND id = ?;',
+				array($user, $graph));
+		}
+
+		header('Location: quicktree.php?location=' . get_nfilter_request_var('location'));
+
+		break;
+	case 'add_ajax':
+		header('Content-type: text/plain');
+
+		print "{ status: 'OK' }";
+
+		break;
```

Comment:
This seems a pointless case?  If not, add some comments as to why it should be kept so other developers aren't confused by it and prevent them accidentally removing it.

Reply:
Yea, I did not look at the console interactions, so I was not certain.  It was there previously.

Reply:
Lets just leave it for now.